File: src/main/java/gmo/cashregister/repositry/admin/RightRepository.java
package gmo.cashregister.repositry.admin;

import gmo.cashregister.entity.security.Right;

import java.util.List;

public interface RightRepository {

    public List<Right> getRights();

    public Right getRight(long id);

    public boolean save(Right right);

    public boolean update(Right right);

    public boolean delete(Right right);
}

File: front/cash-register-admin/src/repositories/RightRepository.js
import {Http} from '../utils/Http'

export class RightRepository {

    newRight() {
        return {
            id: 0,
            code: '',
            name: '',
            description: '',
            api: ''
        };
    }

    async getRights() {
        return await Http.get('admin/rights');
    }

    async getRight(id) {
        return await Http.get('admin/rights/' + id);
    }

    async save(right) {
        return await Http.post('admin/rights', right);
    }

    async update(right) {
        return await Http.put('admin/rights', right);
    }

    async delete(id) {
        return await Http.delete('admin/rights/' + id);
    }
}